node/impl/full/eth.go

```diff
@@ -207,6 +208,56 @@ func (a *EthAPI) FilecoinAddressToEthAddress(ctx context.Context, filecoinAddres
 	return ethtypes.EthAddressFromFilecoinAddress(filecoinAddress)
 }
 
+func (a *EthAPI) FilecoinAddressToEthAddressV1(ctx context.Context, filecoinAddress address.Address, blkParam string) (ethtypes.EthAddress, error) {
+	// If the address is an "f0" or "f4" address, `EthAddressFromFilecoinAddress` will return the corresponding Ethereum address right away.
+	if eaddr, err := ethtypes.EthAddressFromFilecoinAddress(filecoinAddress); err == nil {
+		return eaddr, nil
+	} else if err != ethtypes.ErrInvalidAddress {
+		return ethtypes.EthAddress{}, xerrors.Errorf("error converting filecoin address to eth address: %w", err)
+	}
+
+	// We should only be dealing with "f1"/"f2"/"f3" addresses from here-on.
+	switch filecoinAddress.Protocol() {
+	case address.SECP256K1, address.Actor, address.BLS:
+		// Valid protocols
+	default:
+		// Ideally, this should never happen but is here for sanity checking.
+		return ethtypes.EthAddress{}, xerrors.Errorf("invalid filecoin address protocol: %s", filecoinAddress.String())
+	}
+
+	// Set default block parameter to "finalized" if not provided
+	if blkParam == "" || len(blkParam) == 0 {
+		blkParam = "finalized"
+	}
+
+	// Get the tipset for the specified block
+	ts, err := getTipsetByBlockNumber(ctx, a.Chain, blkParam, true)
+	if err != nil {
+		return ethtypes.EthAddress{}, xerrors.Errorf("failed to get tipset for block %s: %w", blkParam, err)
+	}
+
+	// Lookup the ID address
+	idAddr, err := a.StateManager.LookupIDAddress(ctx, filecoinAddress, ts)
+	if err != nil {
+		return ethtypes.EthAddress{}, xerrors.Errorf(
+			"failed to lookup ID(f0) address for given Filecoin address %s: "+
```

Comment:
it's a `t0` on calibnet, although I'm not sure of the best logic to swap that out here dynamically but I think you can call out to `address.CurrentNetwork`, check whether it's `address.Testnet` or `address.Mainnet` then use either `address.TestnetPrefix` or `address.MainnetPrefix` (although I bet the best way is to do something like `address.NewIDAddress(0)`, `.String()` it and cut off the first 2 chars).